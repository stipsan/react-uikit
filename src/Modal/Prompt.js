import { Component, PropTypes } from 'react'

export default class Prompt extends Component {

  static defaultProps = {
    isOpen: false,
  }

  static propTypes = {
    isOpen: PropTypes.bool,
  }

  componentDidUpdate(prevProps) {
    if (prevProps.isOpen === false && this.props.isOpen === true) {
      this.input.focus()
    }
  }

  render() {
    return (
      <div className="uk-margin-small-top uk-modal-content uk-form">
        <p>
          <input
            className="uk-width-1-1"
            ref={(node) => {
              if (node) {
                this.input = node
              }
            }}
            type="text"
          />
        </p>
      </div>
    )
  }
}
