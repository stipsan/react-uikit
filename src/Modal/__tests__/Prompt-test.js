import React from 'react'
import renderer from 'react-test-renderer'
import Prompt from '../Prompt'

function createNodeMock(element) {
  if (element.type === 'input') {
    return {
      focus() {},
    }
  }
  return null
}

describe('Prompt', () => {
  it('renders correctly', () => {
    const options = { createNodeMock }
    const component = renderer.create(
      <Prompt isOpen={false} />, options
    )

    expect(component.toJSON()).toMatchSnapshot()

    component.update(
      <Prompt isOpen />, options
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
