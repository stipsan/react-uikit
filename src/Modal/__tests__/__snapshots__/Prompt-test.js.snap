// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prompt renders correctly 1`] = `
<div
  className="uk-margin-small-top uk-modal-content uk-form"
>
  <p>
    <input
      className="uk-width-1-1"
      type="text"
    />
  </p>
</div>
`;

exports[`Prompt renders correctly 2`] = `
<div
  className="uk-margin-small-top uk-modal-content uk-form"
>
  <p>
    <input
      className="uk-width-1-1"
      type="text"
    />
  </p>
</div>
`;
